### PR TITLE
dma: fix dma issue on BYT

### PR DESCRIPTION
**Pull request by RanderWang**
It is caused by the change of removing dai waiting after stop.
dma should not be stopped in dai_comp_trigger if HW LLI is not
enable. At this time DMA is still working and it can't be stopped
DMA engine would stop the channel automatically after each transfer.

Signed-off-by: Pan Xiuli <xiuli.pan@linux.intel.com>
Signed-off-by: Rander Wang <rander.wang@linux.intel.com>

### CODE REVIEW

File: src/drivers/dw-dma.c [hunk 1]
```diff
@@ -482,15 +482,17 @@ static int dw_dma_stop(struct dma *dma, int channel)
 #if DW_USE_HW_LLI
 	int i = 0;
 	struct dw_lli2 *lli;
+#else
```
[the remark this answers is not in the document]
Comment (RanderWang) [reply src/drivers/dw-dma.c]: to enable HW LLI on BYT?

[the remark this answers is not in the document]
Comment (lgirdwood) [reply src/drivers/dw-dma.c]: yes

**Comment by xiulipan** [comment]
@lgirdwood 
Enabled HW_LLI on BYT, but the dma could not be stopped, either.
```c
@@ -491,6 +488,9 @@ static int dw_dma_stop(struct dma *dma, int channel)
        dw_write(dma, DW_DMA_CHAN_EN, CHAN_DISABLE(channel));

 #if DW_USE_HW_LLI
+       trace_dma_error("Dli");
+       trace_error_value(dw_read(dma, DW_DMA_CHAN_EN));
+       trace_error_value(channel);
        lli = p->chan[channel].lli;
        for (i = 0; i < p->chan[channel].desc_count; i++) {
                lli->ctrl_hi &= ~DW_CTLH_DONE(1);

```

> 0x10 [211.938786]       delta [211.935215]      dma Dli
> 0x20 [211.938789]       delta [0.000003]        value 0x0000000000000000 
> 0x30 [211.938791]       delta [0.000002]        value 0x0000000000000002 // capture works fine 
> 0x40 [1130.561518]      delta [918.622727]      dma Dli
> 0x50 [1130.561520]      delta [0.000003]        value 0x0000000000000002
> 0x60 [1130.561523]      delta [0.000002]        value 0x0000000000000001 //playback could not stop

**Comment by xiulipan** [comment]
@lgirdwood 
HW LLI for BYT is #365

**Comment by RanderWang** [comment]
> @RanderWang I think on BYT you will need to write NULL to the next + 1 descriptor ptr. This way DMA will load it and see a STOP condition.

@lgirdwood  I have tried this method, still failed. SSP is stopped after DMA stop. So I think DMA pointer does't move without ack from SSP.  And I will go on debugging it.

**Comment by lgirdwood** [comment]
@RanderWang DMA must be stopped before (with DMA FIFO empty and channel in idle) before we stop SSP or other DAIs.

**Comment by RanderWang** [comment]
> @RanderWang DMA must be stopped before (with DMA FIFO empty and channel in idle) before we stop SSP or other DAIs.

Yes, it is just what I described in this patch.  HW LLI can't resolve this issue and I found some regression issues with HW LLI enabled on BYT. My idea is that: the behavior of DMA on BYT is different with APL & CNL, so can we make a special stop function for BYT ? I will update my patch
and any suggestion is welcome

**Comment by RanderWang** [comment]
@lgirdwood  Do you want to enable HW LLI on BYT now?  my patch would base on your choice.

**Comment by lgirdwood** [comment]
@RanderWang @xiulipan yes, make special stop path for BYT with conditional compile. It should only touch dw-dma.c and create a new stop function(s).

**Comment by RanderWang** [comment]
update my patch , thanks for review

**Comment by lgirdwood** [comment]
@RanderWang can you split into 2 patches, one for wait and one for DMA and rebase (for conflicts).

**Comment by RanderWang** [comment]
update my patch, thanks for review

**Comment by lgirdwood** [comment]
@RanderWang checkpatch errors.

**Comment by RanderWang** [comment]
It is line over 80 characters. Do you need me to fix this ?

**Comment by RanderWang** [comment]
Ok, update my patch.

**Pull request merged**